fms/models/mixtral.py

```diff
@@ -0,0 +1,435 @@
+import math
+import re
+from dataclasses import dataclass
+from tkinter import X
+from typing import List, Mapping, Optional, Tuple
+
+import torch
+import torch.nn as nn
+
+from fms import models
+from fms.distributed.strategy import (
+    DistributedStrategy,
+    NoOpStrategy,
+    UniformModelParallelStrategy,
+)
+from fms.modules.attention import MultiHeadAttention
+from fms.modules.feedforward import MOEFeedForward
+from fms.modules.head import LMHead
+from fms.modules.layernorm import LayerNormParameterized
+from fms.modules.positions import RotaryEmbedding
+from fms.utils import serialization
+from fms.utils.config import ModelConfig
+from fms.utils.serialization import FusableWeightsMissingError
+
+
+@dataclass
+class MixtralConfig(ModelConfig):
+    src_vocab_size: int = 32_000  # can be set by tokenizer
+    dim: int = 4096
+    norm_eps: float = 1e-5
+    nheads: int = 32
+    kvheads: int = 8
+    nlayers: int = 32
+    pad_id: int = -1
```

Comment:
another one where I thought we had stopped doing this. this just gets set back to None in the wordembedding and doesn't really mean anything.